rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use crate::{
+    schema::{ans_lookup, current_ans_lookup},
+    utils::{
+        database::PgPoolConnection,
+        util::{
+            bigdecimal_to_u64, deserialize_from_string, parse_timestamp, parse_timestamp_secs,
+            standardize_address,
+        },
+    },
+};
+use aptos_indexer_protos::transaction::v1::{
+    transaction::TxnData, write_set_change::Change as WriteSetChange, DeleteTableItem,
+    Transaction as TransactionPB, WriteTableItem,
+};
+use bigdecimal::BigDecimal;
+use diesel::{prelude::*, ExpressionMethods};
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+use std::{cmp, collections::HashMap};
+
+pub const QUERY_RETRIES: u32 = 5;
+pub const QUERY_RETRY_DELAY_MS: u64 = 500;
+
+type Domain = String;
+type Subdomain = String;
+// Hashmap for looking up previous name records by target address
+pub type AddressToPrimaryNameRecord = HashMap<String, CurrentAnsLookup>;
+// PK of current_ans_lookup, i.e. domain and subdomain name
+pub type CurrentAnsLookupPK = (Domain, Subdomain);
+// PK of ans_lookup, i.e. transaction version, domain, and subdomain
+pub type AnsLookupPK = (i64, Domain, Subdomain);
```

Comment:
You don't need this if you follow the pattern of transaction_version and wsc_index